specs/waku/waku-v2.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Waku
+version: 2.0.0-alpha1
+status: Raw
+authors: Oskar Thorén <oskar@status.im>
+---
+
+## Table of Contents
+
+TODO
+
+## Abstract
+
+Waku is a privacy-preserving peer-to-peer messaging protocol for resource restricted devices. It implements PubSub in libp2p and adds capabilities on top of it. These capabilities are: (i) retrieving historical messages for mostly-offline devices (ii) adaptive nodes, allowing for heterogenerous nodes to contribute, and (iii) bandwidth preservation for light nodes. This makes it ideal for running a p2p protocol on mobile.
+
+Historically, it has its roots in Waku v1 [xx5], which stems from Whisper [xx6], originally part of the Ethereum stack. However, Waku v2 acts more as a thin wrapper for PubSub and has a different API. It is implemented in an iterative manner where initial focus is on porting essential functionality to libp2p. See here for a rough road map [xx7].
```

Comment:
Maybe link both waku v1 and whisper?

Reply:
They are linked at the bottom of the document. Will do formatting later, this is just to capture raw state. Lets focus on content and shape of spec first please.

Reply:
I fixed this anyway now